File: app/src/main/java/com/ShaHar91/poems/ui/base/normal/BaseGoogleViewModel.java
package com.shahar91.poems.ui.base.normal;

import androidx.lifecycle.ViewModel;

import java.util.regex.Pattern;

import io.reactivex.disposables.CompositeDisposable;

/**
 * This is the ViewModel that will be used with every Activity
 */
public abstract class BaseGoogleViewModel extends ViewModel {
    private final CompositeDisposable compositeDisposable = new CompositeDisposable();

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.dispose();
    }

    public boolean checkDataValidity(String inputText, Pattern pattern) {
        return inputText.trim().isEmpty() || !pattern.matcher(inputText).matches();
    }
}

File: app/src/main/java/com/ShaHar91/poems/ui/base/BaseGoogleViewModel.kt
package com.shahar91.poems.ui.base

import androidx.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable
import java.util.regex.Pattern

/**
 * This is the ViewModel that will be used with every Activity
 */
abstract class BaseGoogleViewModel : ViewModel() {
    private val compositeDisposable = CompositeDisposable()
    override fun onCleared() {
        super.onCleared()
        compositeDisposable.dispose()
    }

    fun checkDataValidity(inputText: String, pattern: Pattern): Boolean {
        return inputText.trim().isEmpty() || !pattern.matcher(inputText).matches()
    }
}